packages/lit-html/src/static.ts

```diff
@@ -0,0 +1,77 @@
+import {html as coreHtml, svg as coreSvg, TemplateResult} from './lit-html.js';
+
+/**
+ * Wraps a string so that it behaves like part of the static template
+ * strings instead of a dynamic value.
+ *
+ * This is a very unsafe operation and may break templates if changes
+ * the structure of a template. Do not pass user input to this function
+ * without sanitizing it.
+ */
+export const unsafeStatic = (value: string) => ({
+  _$litStatic$: value,
+});
+
+type StaticValue = ReturnType<typeof unsafeStatic>;
+
+const stringsCache = new WeakMap<TemplateStringsArray, TemplateStringsArray>();
+
+/**
+ * Generates a template literal tag function that returns a TemplateResult with
+ * the given result type.
+ */
+const tag = (coreTag: typeof coreHtml) => (
+  strings: TemplateStringsArray,
+  ...values: unknown[]
+): TemplateResult => {
+  let processedStrings = stringsCache.get(strings);
```

Comment:
If the static values change, we won't get a different `processedStrings`. Didn't we decide to include the static values as part of the template key?

Reply:
Fixed